state-chain/pallets/cf-lending-pools/src/chp_lending.rs

```diff
@@ -301,7 +316,8 @@ pub fn chp_upkeep<T: Config>(current_block: BlockNumberFor<T>) -> Weight {
 	}
 
 	// Calculate final weight
-	T::WeightInfo::upkeep_active(count_active) +
+	weight_consumed +
+		T::WeightInfo::upkeep_active(count_active) +
```

Comment:
There doesn't seem to be a significant difference in weight between different types of loan states (active/liquidating/finalising), so I think we could simplify and just use the "no_action" one. Then IMO the logic for how we compute the weight would be more clear: we use "no_action" as the baseline and add weight of the actions that we do perform.